React/Base/RCTEventDispatcher.m

```diff
@@ -28,12 +29,15 @@
   return eventName;
 }
 
+// Key generator for events that cannot be coalesced.
+static uint16_t RCTNonCoalescableEventKey = 0;
+
 static NSNumber *RCTGetEventID(id<RCTEvent> event)
 {
   return @(
     event.viewTag.intValue |
     (((uint64_t)event.eventName.hash & 0xFFFF) << 32) |
-    (((uint64_t)event.coalescingKey) << 48)
+    (((uint64_t)([event canCoalesce] ? event.coalescingKey : RCTNonCoalescableEventKey++)) << 48)
```

Comment:
We access optional `canCoalesce` and `coalescingKey` here.
So, we have two options here:
 1. Do `responseToSelector` check here.
 2. Make that props required for all implementations.

I strongly prefer 2 because we should maintain this function as perfromant as possible.

Right?

Reply:
Oh, no, I see, I am wrong.